android/src/main/java/com/brentvatne/common/react/VideoEventEmitter.kt

```diff
@@ -48,7 +49,8 @@ enum class EventTypes(val eventName: String) {
         fun toMap() =
             mutableMapOf<String, Any>().apply {
                 EventTypes.values().toList().forEach { eventType ->
-                    put("top${eventType.eventName.removePrefix("on")}", mapOf("registrationName" to eventType.eventName))
+                    put(eventType.eventName, MapBuilder.of("registrationName", eventType.eventName))
+                    put("top${eventType.eventName.removePrefix("on")}", MapBuilder.of("registrationName", eventType.eventName))
```

Comment:
We can go with this if
```suggestion
                   if (BuildConfig.IS_NEW_ARCHITECTURE_ENABLED)
                        put("top${eventType.eventName.removePrefix("on")}", mapOf("registrationName" to eventType.eventName))
                   else
                        put(eventType.eventName, MapBuilder.of("registrationName", eventType.eventName))
```

If this change fix the issue I am fine to merge this

Reply:
I agree with your suggestion, but it is a bit more complicated :)
We need to use the isNewArchEnabled from Application, then we need context to retrieve it. but yes this is what I had in mind !

Reply:
This is loaded from Application you can see in `build.gradle`
```
def isNewArchitectureEnabled() {
    return rootProject.hasProperty("newArchEnabled") && rootProject.getProperty("newArchEnabled") == "true"
}
```
we are getting this property from `rootProject` (app) so this should be fine

Reply:
Sorry for late response.
AFAIK, contional code is not required because if "top" is not a prefix, React Native change it internally.
Also [this code](https://github.com/TheWidlarzGroup/react-native-video/blob/v6.4.2/android/src/main/java/com/brentvatne/common/react/VideoEventEmitter.kt#L283) should be change for event work properly.

nit; facebook MapBuilder is meaningless it just hashMap wrapper. we can replace it with just use `mapof`

Reply:
Ok, regarding to the code you are pointing, the  `override fun getEventName() = "top${event.eventName.removePrefix("on")}" ` vs `override fun getEventName() = "${event.eventName}" ` should depend on newArch ?

Reply:
@freeboub 
No. This part needs to be handled so that it syncs with the value returned by `ReactExoplayerViewManager.getExportedCustomDirectEventTypeConstants`. This is architecture independent.

Reply:
I haven't tested it yet, but it seems like a simple solution would be to remove all the parts that add the top prefix.